### ISSUE TITLE
IllegalStateException in FiltersAggregatorTests.testFilterByFilterCost

### ISSUE THREAD

**Issue by not-napoleon** [elastic/elasticsearch#67835]
**Build scan**:
https://gradle-enterprise.elastic.co/s/kpzbc6uteucmg/tests/:server:test/org.elasticsearch.search.aggregations.bucket.filter.FiltersAggregatorTests/testFilterByFilterCost#1

**Repro line**:

REPRODUCE WITH: 
```
./gradlew ':server:test' --tests "org.elasticsearch.search.aggregations.bucket.filter.FiltersAggregatorTests.testFilterByFilterCost" \
  -Dtests.seed=3CB467C616458C42 \
  -Dtests.security.manager=true \
  -Dtests.locale=he-IL \
  -Dtests.timezone=Africa/Timbuktu \
  -Druntime.java=8
```

**Reproduces locally?**:
Yes.

**Applicable branches**:
7.11

**Failure history**:

https://gradle-enterprise.elastic.co/scans/tests?search.relativeStartTime=P7D&search.timeZoneId=America/New_York&tests.container=org.elasticsearch.search.aggregations.bucket.filter.FiltersAggregatorTests&tests.sortField=FAILED&tests.test=testFilterByFilterCost&tests.unstableOnly=true

**Failure excerpt**:
java.lang.IllegalStateException: unexpected docvalues type NONE for field 'test' (expected one of [SORTED_NUMERIC, NUMERIC]). Re-index with correct docvalues type.

**Comment by not-napoleon** [comment]
It repros on 7.11 but not 7.x, so maybe just something needs to be backported?

**Comment by nik9000** [comment]
> It repros on 7.11 but not 7.x, so maybe just something needs to be backported?

Thanks for calling that out! It'll help a ton.

**Comment by nik9000** [comment]
I fixed by backporting a1ca18a  to 7.11.
